pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -150,8 +160,15 @@ func NewStaticPolicy(topology *topology.CPUTopology, numReservedCPUs int, reserv
 		return nil, err
 	}
 
-	klog.InfoS("Reserved CPUs not available for exclusive assignment", "reservedSize", reserved.Size(), "reserved", reserved)
+	reservedPhysicalCPUs := topology.ExpandCPUsWithCoreSiblings(reserved)
+	if reservedPhysicalCPUs.Size() < reserved.Size() {
```

Comment:
Just out of curiosity, when can we run into this case?  Since `reservedPhysicalCPUs` is a union of reserved `coreCPUSiblings` it should always be >= reserved size. Are we trying to be overcautious here?

Reply:
good point! this can't fail unless the sky is falling (or we have bugs we should have caught elsewhere). Let me remove this code.